lib/endpoints/class-wp-rest-comments-controller.php

```diff
@@ -640,6 +640,18 @@ protected function prepare_links( $comment ) {
 			);
 		}
 
+		// Only grab one comment to verify the comment has children.
+		$comment_children = $comment->get_children( array( 'number' => 1 ) );
```

Comment:
In code review @rmccue identified this to be a performance concern, as `get_children` runs a query every time that is not object-cached. Performance impact needs to be assessed on a site with a high volume of comments before this can be merged.